cmd/operator/main.go

```diff
@@ -88,6 +88,7 @@ func main() {
 		evaluatorMemoryResource = flag.Int64("evaluator-memory-resource", 200, "The Memory Resource of evaluator pod, in mega bytes.")
 		evaluatorMemoryLimit    = flag.Int64("evaluator-memory-limit", 1000, "The Memory Limit of evaluator pod, in mega bytesv.")
 		evaluatorCPUResource    = flag.Int64("evaluator-cpu-resource", 100, "The CPU Resource of evaluator pod, in milli cpu.")
+		operatorDeployment      = flag.String("opeator-deployment", "kubectl-deployed", "the environment operator runs on.")
```

Comment:
typo: `opeator`. Also the flag description should say something like "how the operator is deployed".

Naming thought: to avoid overloading the term deployment (e.g. k8s resource), what about, just `mode` or `strategy`?

Also, what do you think about asserting the flag to be an enum in code to prevent abuse? e.g.

```go

mode = flag.String("mode", "kubectl", "how managed collection was provisioned")

switch mode {
// repo manifest always defaults to "kubectl".
case "kubectl":
case "gcloud":
case "gcloud-auto":
default:
	logger.Error(err, "--mode must be one of {'kubectl', 'gcloud', 'gcloud-auto'}")
	os.Exit(1)
}
```

@lyanco any thoughts on naming the enums here? AFAICT from this change, user-agent would be something like (i think this is all combinations):
- `prometheus-engine-export/0.3.3 prometheus-collector/2.28.1-gmp.7 (mode:kubectl)`
- `prometheus-engine-export/0.3.3 prometheus-collector/2.28.1-gmp.7 (mode:gcloud)`
- `prometheus-engine-export/0.3.3 prometheus-collector/2.28.1-gmp.7 (mode:gcloud-auto)`
- `prometheus-engine-export/0.3.3 prometheus/2.28.1-gmp.7`
- `prometheus-engine-export/0.3.3 rule-evaluator/0.3.3 (mode:kubectl)`
- `prometheus-engine-export/0.3.3 rule-evaluator/0.3.3 (mode:gcloud)`
- `prometheus-engine-export/0.3.3 rule-evaluator/0.3.3 (mode:gcloud-auto)`
- `prometheus-engine-export/0.3.3 rule-evaluator/0.3.3`

Reply:
I'd probably replace "gcloud" with "gke" but besides that, looks great to me

Reply:
done